src/renderer/views/Channel/Channel.vue

```diff
@@ -39,6 +39,14 @@
         :related-channels="relatedChannels"
       />
       <div class="select-container">
+        <FtToggleSwitch
+          v-if="(currentTab === 'videos') || (currentTab === 'shorts') || (currentTab === 'live')"
+          v-show="!isElementListLoading"
+          v-model="hideWatchedToggle"
```

Comment:
FtToggleSwitch doesn't support v-model, you've not noticed it until now because you aren't actually tracking the value of the switch.
```suggestion
          :default-value="hideWatchedToggle"
```